src/compas_view2/app/controller.py

```diff
@@ -31,18 +32,29 @@ class Controller:
     def __init__(self, app):
         self.app = app
 
-    def interactive(func):
-        def wrapped(self):
-            def add(obj):
-                if obj:
-                    self.app.add(obj)
-                    self.app.view.update()
-
-            worker = Worker(func, self)
-            worker.signals.result.connect(add)
-            Worker.pool.start(worker)
-        return wrapped
-
+    def interactive(action="add"):
+        def outer(func):
+            def wrapped(self):
+                def add(obj):
+                    if obj:
+                        self.app.add(obj)
+                        self.app.view.update()
+
+                def edit(obj):
+                    if obj:
+                        self.editform = PointEditForm(obj)
```

Comment:
This is temporary here, it will be replaced by the general `EditForm` class, which detects the type of object through registrations.

Reply:
was wondering if we could make the scene objects into data classes and have the forms created on the fly based on the data attribute types of the objects?

or is that overkill?